services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -569,21 +570,52 @@ def __init__(self, *args: object, num_rows: int, row_group_byte_size: int) -> No
         self.row_group_byte_size = row_group_byte_size
 
 
-def get_parquet_file_and_size(url: str, hf_endpoint: str, hf_token: Optional[str]) -> tuple[pq.ParquetFile, int]:
+def open_file(file_url: str, hf_endpoint: str, hf_token: Optional[str]) -> HfFileSystemFile:
     fs = HfFileSystem(endpoint=hf_endpoint, token=hf_token)
-    f = fs.open(url)
-    return pq.ParquetFile(f), f.size
+    return fs.open(file_url)
 
 
-def retry_and_validate_get_parquet_file_and_size(
+def retry_validate_get_parquet_file_and_size(
     url: str, hf_endpoint: str, hf_token: Optional[str], validate: Optional[Callable[[pq.ParquetFile], None]]
 ) -> tuple[pq.ParquetFile, int]:
+    """
+    Get parquet file as a pq.ParquetFile at a given url, and its size in bytes.
+    Also validate parquet file if validation function is passed with `validate` argument.
+
+    Returns:
+        `tuple[pq.ParquetFile, int]` - (parquet files, size in bytes)
+    """
     try:
-        sleeps = [0.2, 1, 1, 10, 10, 10]
-        pf, size = retry(on=[pa.ArrowInvalid], sleeps=sleeps)(get_parquet_file_and_size)(url, hf_endpoint, hf_token)
+        f = retry(on=[pa.ArrowInvalid], sleeps=SLEEPS)(open_file)(url, hf_endpoint, hf_token)
+        pf, size = pq.ParquetFile(f), f.size
         if validate:
             validate(pf)
         return pf, size
```

Comment:
should `f` be closed?

Reply:
ideally yes (as in second func) but in this case it would cause `operation on closed file` error after

Reply:
i can do like this [82bf968](https://github.com/huggingface/datasets-server/pull/2564/commits/82bf968e7e3f2a6adcdd42416f7e595fae5ba459)
cc @lhoestq